File: QytetetRuby/lib/sorpresa.rb
#encoding: utf-8
module ModeloQytetet
  class Sorpresa
    attr_accessor :texto, :valor, :tipo

    def initialize(texto, valor, tipo)
      @texto = texto
      @valor = valor
      @tipo = tipo
    end

    def get_texto
      @texto
    end

    def get_valor
      @valor
    end

    def get_tipo
      @tipo
    end

    def to_s
      "Texto: #{@texto} \n Valor: #{@valor} \n Tipo: #{@tipo}"
    end

  end
end

File: QytetetJava/src/modeloqytetet/Sorpresa.java

package modeloqytetet;

import java.util.ArrayList;

public class Sorpresa {
    private String texto;
    private TipoSorpresa tipo;
    private int valor;
    //private static ArrayList<Sorpresa> mazo;
    
    public Sorpresa(String texto, int valor, TipoSorpresa tipo) {
        this.texto = texto;
        this.tipo = tipo;
        this.valor = valor;
    }
    
    public String getTexto() {
        return texto;
    }
    
    public int getValor() {
        return this.valor;
    }
    
    public TipoSorpresa getTipo() {
        return this.tipo;
    }
        
    @Override
    public String toString() {
        return "\nSorpresa{" + "texto=" + texto + ", valor=" + Integer.toString(valor) + ", tipo=" + tipo + "}";
    }
}
